app/utils/Manager.ts

```diff
@@ -33,52 +43,58 @@ export default class Manager extends Internal {
         return this.managerData?.version?.edition === Consts.EDITION.COMMUNITY;
     }
 
-    getManagerUrl(url, data?) {
-        return this.buildActualUrl(url, data);
-    }
+    getManagerUrl: Manager['buildActualUrl'] = (url, params) => {
+        return this.buildActualUrl(url, params);
+    };
 
     getSelectedTenant() {
         return this.managerData?.tenants?.selected ?? null;
     }
 
     getSystemRoles() {
         const roles = this.managerData?.roles ?? null;
-        return _.filter(roles, role => role.type === 'system_role');
+        return filter(roles, role => role.type === 'system_role');
     }
 
-    // eslint-disable-next-line class-methods-use-this
-    buildActualUrl(url, data?) {
-        const path = `/sp${getUrlWithQueryString(url, data)}`;
+    buildActualUrl: typeof getUrlWithQueryString = (url, params) => {
+        const path = `/sp${getUrlWithQueryString(url, params)}`;
         return StageUtils.Url.url(path);
-    }
-
-    doFetchFull(fetcher, params = {}, fullData = { items: [] }, size = 0) {
+    };
+
+    doFetchFull<ResponseBodyItem>(
+        fetcher: (params: QueryStringParams) => Promise<PaginatedResponse<ResponseBodyItem>>,
+        params: QueryStringParams = {},
```

Comment:
Optional:

You could probably extend that generic type to get second one and be something like `doGet` in `External` class:

```suggestion
    doFetchFull<ResponseBodyItem, RequestQueryParams extends QueryParams = QueryParams>(
        fetcher: (params: RequestQueryParams) => Promise<PaginatedResponse<ResponseBodyItem>>,
        params: RequestQueryParams = {},
```

Similar approach can be taken to `doPostFull`.

I didn't test it. 

It's definitely not a blocker.

Reply:
As for now I resigned from the idea of passing a type for `params` (92f574de16f56908de28bce85331c503a1a0f3b5) I wouldn't push it within those changes, to keep things consistent 💪 

But idea seems to be interesting 🎉